src/include/sof/trace/trace.h

```diff
@@ -170,6 +170,34 @@ void trace_log(bool send_atomic, const void *log_entry,
 unsupported_amount_of_params_in_trace_event\
 _thrown_from_macro_BASE_LOG_in_trace_h
 
+#define trace_check_size_uint32_0() do { } while (0)
+
+#define trace_check_size_uint32_1(a) do {				\
+	STATIC_ASSERT(sizeof(a) <= sizeof(uint32_t),			\
+	trace_parameter_too_large);					\
+} while (0)
+
+#define trace_check_size_uint32_2(a, b) do {				\
+	  trace_check_size_uint32_1(a);					\
+	  trace_check_size_uint32_1(b);					\
+} while (0)
+
+#define trace_check_size_uint32_3(a, b, c) do {				\
+	  trace_check_size_uint32_1(a);					\
+	  trace_check_size_uint32_1(b);					\
+	  trace_check_size_uint32_1(c);					\
+} while (0)
+
+#define trace_check_size_uint32_4(a, b, c, d) do {			\
+	  trace_check_size_uint32_1(a);					\
+	  trace_check_size_uint32_1(b);					\
+	  trace_check_size_uint32_1(c);					\
+	  trace_check_size_uint32_1(d);					\
+} while (0)
+
+#define trace_check_args(...) META_CONCAT(trace_check_size_uint32_,	\
+	META_COUNT_VARAGS_BEFORE_COMPILE(__VA_ARGS__))(__VA_ARGS__)
+
```

Comment:
@lyakh maybe we would like to use META_MAP(), then trace_check_size_uint32_2, trace_check_size_uint32_3, trace_check_size_uint32_4 could be removed and solution will be more flexible when possible number of max number of trace arguments change.

Reply:
@ktrzcinx you mean like
```
#define trace_check_size_uint32_2(a, b) META_MAP(1, trace_check_size_uint32_1, a, b)
#define trace_check_size_uint32_3(a, b, c) META_MAP(1, trace_check_size_uint32_1, a, b, c)
#define trace_check_size_uint32_4(a, b, c, d) META_MAP(2, trace_check_size_uint32_2, a, b, c, d)
```
yeah, sure, why not make it a bit more confusing yet :-D Or just make the above a bit more compact by
```
#define trace_check_size_uint32_3(a, b, c) do {				\
	  trace_check_size_uint32_2(a, b);				\
	  trace_check_size_uint32_1(c);					\
} while (0)

#define trace_check_size_uint32_4(a, b, c, d) do {			\
	  trace_check_size_uint32_2(a, b);				\
	  trace_check_size_uint32_2(c, d);				\
} while (0)
```

Reply:
I think rather about something like:
```
#define trace_check_size_uint32_1(a)
	STATIC_ASSERT(sizeof(a) <= sizeof(uint32_t), trace_parameter_too_large);

#define STATIC_CHECK_ARGS(...) do { META_RECURSE( \
	META_MAP(1, META_DEFER(trace_check_size_uint32_1), __VA_ARGS__)); \
	} while (0)

#define _log_message(atomic, lvl, comp_class, ctx, id_1, id_2,		\
		     format, ...)					\
do {									\
	_DECLARE_LOG_ENTRY(lvl, format, comp_class,			\
			   PP_NARG(__VA_ARGS__));			\
	STATIC_CHECK_ARGS(__VA_ARGS__);					\
	STATIC_ASSERT(
        ...
```
*not tested

Reply:
@ktrzcinx I tried that and it didn't compile. Maybe I'm using it wrongly, and I don't see any other uses of `META_MAP()` in the code, so, unless you volunteer to investigate, maybe it's better to stay with the current solution for now.

Reply:
Given that we are close to v1.6 lets merge this today unless @ktrzcinx can update the macro as it's better than what we have today. We can always revisit after v1.6. I would also like us to revisit ```META_MAP()``` if there are no users.

Reply:
It wasn't one line change, so I created separate PR covering this topic: https://github.com/thesofproject/sof/pull/3393.

Reply:
Great - @lyakh shall we close this one ?

Reply:
@lgirdwood sure, would @ktrzcinx also pull the third patch from this series into his PR?

Reply:
done